core/src/main/java/org/elasticsearch/common/settings/SettingsModule.java

```diff
@@ -200,4 +203,16 @@ private void validateTribeSettings(Settings settings, ClusterSettings clusterSet
             }
         }
     }
+
+    public Settings getSettings() {
+        return settings;
+    }
+
+    public IndexScopedSettings getIndexScopedSettings() {
+        return indexScopedSettings;
+    }
+
+    public ClusterSettings getClusterSettings() {
```

Comment:
Nit, and nothing to do with this PR but just something I noticed: we should make this terminology clearer, for example either using "scoped" in all of these or none of them? And perhaps having "node" in the first one, like getNodeSettings()? Just an idea for future work.